File: src/main/java/pl/uwr/server/model/Reservation.java
package pl.uwr.server.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.Calendar;

@NoArgsConstructor
@Entity
public class Reservation {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private Long classId;
    private String title;
    private Calendar beginDate;
    private Calendar endDate;
    @ManyToOne
    @JsonIgnore
    private Student student;
    @Column(name = "active")
    private boolean isActive;

    public Reservation(Long classId, String title, Calendar beginDate, Calendar endDate, Student student) {
        this.classId = classId;
        this.title = title;
        this.beginDate = beginDate;
        this.endDate = endDate;
        this.student = student;
        isActive = true;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getClassId() {
        return classId;
    }

    public void setClassId(Long classId) {
        this.classId = classId;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public Calendar getBeginDate() {
        return beginDate;
    }

    public void setBeginDate(Calendar beginDate) {
        this.beginDate = beginDate;
    }

    public Calendar getEndDate() {
        return endDate;
    }

    public void setEndDate(Calendar endDate) {
        this.endDate = endDate;
    }

    public Student getStudent() {
        return student;
    }

    public void setStudent(Student student) {
        this.student = student;
    }

    public boolean isActive() {
        return isActive;
    }

    public void setActive(boolean active) {
        isActive = active;
    }
}

File: src/main/java/pl/uwr/server/model/Reservation.kt
package pl.uwr.server.model

import com.fasterxml.jackson.annotation.JsonIgnore
import lombok.NoArgsConstructor
import java.util.*
import javax.persistence.*

@NoArgsConstructor
@Entity
/**
 * klasa reprezentująca obiekty przechowywane w repozytorium rezerwacji
 * @param classId numer sali
 * @param title tytuł rezerwacji
 * @param beginDate termin początku rezerwacji
 * @param endDate temin końca rezerwacji
 */
class Reservation(var classId: Long,
                  var title: String,
                  var beginDate: Calendar,
                  var endDate: Calendar,
                  @JsonIgnore @ManyToOne var student: Student) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
            /**
             * ID rezerwacji
             */
    var id: Long = 0L
    @Column(name = "active")
            /**
             * czy rezerwacja jest aktywna
             */
    var isActive = true
    /**
     * opcjonalne pełne imię i nazwisko rezerwującego. Wykorzystywane przy rezerwacjach cyklicznych
     */
    val studentFullName = ""
}